Copy Edit for User Manual (#65)
KeychainMDIP/keychain-cli: doc/CLI-user-manual/README.md

>   help [command]                              display help for command

Author: Updated in 69becfd482733c11287bd313d2581aba05bc0f3a

```

</details>

## Quick Start

The following examples use a `$` to denote the shell prompt:

```sh
$ kc
```

> !!! note Unless you edit your shell's `$PATH` variable, you need to invoke kc with a `./` prefix to run the script in the current directory:

Author: The Markdown spec does not include a callout. The precious form worked in some renderers but not GitHub. Changed to a format GFM likes, and we'll update again when we create a site to serve the docs. d1693400dd2f8a1396626c76cd74e40fba2254f2

> !!! note Unless you edit your shell's `$PATH` variable, you need to invoke kc with a `./` prefix to run the script in the current directory:

Reviewer: This is exactly what I was hoping for

### Backing Up and Recovering a Wallet

To recover a wallet from a backup, you need two pieces of information:
- the seed phrase, aka mnemonic,
- the backup DID.

The mnemonic consists of 12 short words (BIP-39) that are used to generate the wallet's private keys:

```sh
$ kc show-mnemonic
know soon mind pen polar pulse patient salmon wage friend equip rotate
```

Creating a wallet backup encrypts the current state of a user wallet content in a DID Document:

```sh
$ kc backup-wallet
did:mdip:test:z3v8Auairrc7XjSdoA1QvuytZXmGdmjcaFsPb2xKjM6TzowPKRn
```

If you lose the wallet file, you can regenerate the private keys from the mnemonic:

> !!! danger import-wallet will overwrite an existing wallet!

Reviewer: ```suggestion
```

This was fixed, so this warning can be removed

Author: Removed in 7762b1350da82734ccf51d4431c5aefebd009a55

            "https://www.w3.org/ns/did/v1"
        ],
        "id": "did:mdip:test:z3v8AuabRm9DaiakqbwFPgsLd6vSYBQtdj7poQFGYBgsZCfqTvY",

    ...

    "didDocumentData": {
        "vault": "did:mdip:test:z3v8AuafhKoRuEkDTjyoabgPXKx4Yi4cPmPdzUgMNyKxkzYNA6u"
    },
    
    ...

}
```

Note that each wallet and each identity have their own backups. This will allow the user to chose a different MDIP registry (or no registry) with different security or permanence attributes for a particular identity (ie: some DIDs will be more valuable than others to the user).

### Removing an ID

At any time, a user may remove a named DID from their wallet:

```sh
$ kc remove-id Alice
ID Alice removed
```

### Recovering an ID

Recovery of a DID's history using the Vault DID is possible because the Vault data is encrypted with the wallet's keys. The wallet keys are used to decrypt the Vault DID data containing the DID's private history:

```sh
$ kc recover-id did:mdip:test:z3v8AuabRm9DaiakqbwFPgsLd6vSYBQtdj7poQFGYBgsZCfqTvY

> Recovered Alice!

Author: I brought this up in the extended commit message introducing these. My thinking was that we'd add highlighting for the most relevant bits for now, and update to separate command from output once we had a SSG with prism rendering them. But for now we can just drop this commit.

> Recovered Alice!

Reviewer: OK to leave in for now

>
> \- [Source](https://www.w3.org/TR/vc-data-model/#abstract)

### What is a Verifiable Credential?

From W3C:
> A verifiable credential (VC) can represent all of the same information that a physical credential represents. The addition of technologies, such as digital signatures, makes verifiable credentials more tamper-evident and more trustworthy than their physical counterparts.
>
> \- [Source](https://www.w3.org/TR/vc-data-model/#what-is-a-verifiable-credential)

### MDIP Verifiable Credential Basic Workflow

![](workflow.png)

The basic workflow involves three actors: Alice (the Issuer), Bob (the Holder), and Carol (the Verifier). In this scenario, Bob wishes to gain access to some resource controlled by Carol. Carol will grant Bob access only if Bob can prove that he owns a particular credential issued by Alice.

#### Steps to Creating a VC:

> 1. The Issuer (Alice) creates a Credential. The Credential is like a type or a class; it describes the Credential, and specifies a schema. Each Verified Credential is an instance of a Credential.

Author: I _think_ the most accurate phrasing would be "...creates a DID of the Credential type", or "...creates a DID using the Credential schema". With "type" being an adjective here, we need a noun word for it to apply to.

> 1. The Issuer (Alice) creates a Credential. The Credential is like a type or a class; it describes the Credential, and specifies a schema. Each Verified Credential is an instance of a Credential.

Reviewer: OK TBD

1. The Issuer binds a Credential to a Holder
1. The Issuer attest a Credential, creating a Verifiable Credential (VC),  by signing and encrypting a Bound Credential.
1. The Holder (Bob) accepts the VC (adding it to their wallet for future use).
1. The Verifier (Carol) creates a Challenge. A Challenge is a list of Credentials and trusted Issuers.
1. The Holder creates a Response to the Challenge. A Response contains a list of Verified Presentations that correspond to the Credentials and trusted Issuers listed in the Challenge. If the User's wallet contains the right Verified Credentials, then it will be possible to create a valid Response to the Challenge. The Response is encrypted for the Verifier.

> 1. The Verifier verifies the Response. The Response is validated by checking that it contains a Verified Presentation for every Credential in the Challenge, and that the VP is issued by a trusted Issuer, and that the VC corresponding the VP has not been revoked. If everything checks out the Response is accepted and the Holder is granted authorization.

Author: Yes. In fact, I'd like to use it more to avoid redundant word use on L554:

```suggestion
1. The Verifier validates the Response. The Response is verified by checking that it contains a Verified Presentation for every Credential in the Challenge,  that the VP is issued by a trusted Issuer, and that the VC corresponding the VP has not been revoked. If everything checks out the Response is accepted and the Holder is granted authorization.
```

> The DID Manifest is a public data container that is returned with the DID document metadata. Users can chose to publish or reveal any attestation they receive. All information in the manifest is publicly viewable.

Reviewer: ```suggestion
The DID Manifest is a public data container that is returned with the DID document data. Users can chose to publish or reveal any attestation they receive. All information in the manifest is publicly viewable.
```

### Encrypting a message

The MDIP CLI can be used to encrypt a message that can only be decrypted by the targeted DID:

```sh
$ kc encrypt-msg 'this is another secret message' david
did:mdip:z3v8AuacNPvBNSN8o1LgJxSD9jZVQBkre8BfHdrPgSugb7zuhqs
```

### Decrypting a message or a file

Recipients of MDIP encrypted messages or files can use the command below to decrypt the received content:

```sh
$ kc decrypt-did did:mdip:z3v8AuacNPvBNSN8o1LgJxSD9jZVQBkre8BfHdrPgSugb7zuhqs
this is another secret message
```

### Decrypting JSON from a VC

> All other MDIP documents are also encrypted. The command below is used to decrypt content from VCs and other documents:

Reviewer: ```suggestion
Some MDIP documents such as VCs are encrypted JSON. This command command combines decrypting with parsing as JSON:
```

Author: Edited your suggestion for a repeated word, hope that's kosher :smile:
